File: software/examples/csharp/ExampleInterrupt.cs
using System;
using Tinkerforge;

class Example
{
	private static string HOST = "localhost";
	private static int PORT = 4223;
	private static string UID = "XYZ"; // Change XYZ to the UID of your IO-16 Bricklet 2.0

	// Callback function for input value callback
	static void InputValueCB(BrickletIO16V2 sender, byte channel, bool changed,
	                         bool value)
	{
		Console.WriteLine("Channel: " + channel);
		Console.WriteLine("Changed: " + changed);
		Console.WriteLine("Value: " + value);
		Console.WriteLine("");
	}

	static void Main()
	{
		IPConnection ipcon = new IPConnection(); // Create IP connection
		BrickletIO16V2 io = new BrickletIO16V2(UID, ipcon); // Create device object

		ipcon.Connect(HOST, PORT); // Connect to brickd
		// Don't use device before ipcon is connected

		// Register input value callback to function InputValueCB
		io.InputValueCallback += InputValueCB;

		// Set period for input value (channel 4 [A4]) callback to 0.5s (500ms)
		io.SetInputValueCallbackConfiguration(4, 500, false);

		Console.WriteLine("Press enter to exit");
		Console.ReadLine();
		ipcon.Disconnect();
	}
}

File: software/examples/rust/example_interrupt.rs
use std::{io, error::Error};
use std::thread;
use tinkerforge::{ip_connection::IpConnection, 
                  io16_v2_bricklet::*};


const HOST: &str = "localhost";
const PORT: u16 = 4223;
const UID: &str = "XYZ"; // Change XYZ to the UID of your IO-16 Bricklet 2.0.

fn main() -> Result<(), Box<dyn Error>> {
    let ipcon = IpConnection::new(); // Create IP connection.
    let io = Io16V2Bricklet::new(UID, &ipcon); // Create device object.

    ipcon.connect((HOST, PORT)).recv()??; // Connect to brickd.
    // Don't use device before ipcon is connected.

     let input_value_receiver = io.get_input_value_callback_receiver();

        // Spawn thread to handle received callback messages. 
        // This thread ends when the `io` object
        // is dropped, so there is no need for manual cleanup.
        thread::spawn(move || {
            for input_value in input_value_receiver {           
                		println!("Channel: {}", input_value.channel);
		println!("Changed: {}", input_value.changed);
		println!("Value: {}", input_value.value);
		println!();
            }
        });

		// Set period for input value (channel 4) callback to 0.5s (500ms).
		io.set_input_value_callback_configuration(4, 500, false);

    println!("Press enter to exit.");
    let mut _input = String::new();
    io::stdin().read_line(&mut _input)?;
    ipcon.disconnect();
    Ok(())
}
